akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/Effect.scala

```diff
@@ -230,6 +233,64 @@ object Effect {
 
   final case class TimerCancelled(key: Any) extends Effect
 
+  /**
+   * The behavior initiated an ask of a targeted recipient.
+   */
+  final class AskInitiated[Req, Res, T](
+      val target: RecipientRef[Req],
+      val responseClass: Class[Res],
+      createMessage: ActorRef[Res] => Req,
+      mapResponse: Try[Res] => T)
```

Comment:
I think this should rather be built in and be a `send/runResponse(T): Unit` just like `send` on the `TimerScheduled` - failing to map should happen inside actor logic like it will in a real running actor. Will require some stateful machinery inside the testkit since the same ask-response actorref can only be used once.

Since an ask can only fail with timeout, I don't think we should allow sending arbitrary `Failure`'s so perhaps a separate method for that, something like `triggerResponseTimeout()` ?